src/msix/PAL/Signature/Win32/SignatureValidator.cpp

```diff
@@ -351,29 +360,40 @@ namespace MSIX
             ),"CryptQueryObject failed");
         unique_cert_store_handle certStoreHandle(certStoreHandleT);
 
-        PCCERT_CONTEXT pCertContext = NULL;
         if (dwContentType == CERT_QUERY_CONTENT_CERT)
         {   //get the certificate context
-            pCertContext = CertEnumCertificatesInStore(certStoreHandle.get(), NULL);
+            return CertEnumCertificatesInStore(certStoreHandle.get(), NULL);
         }
         else 
         {   //pkcs7 -- get the end entity
+            PCCERT_CONTEXT pCertContext = NULL;
             while (NULL != (pCertContext = CertEnumCertificatesInStore(certStoreHandle.get(), pCertContext)))
-            {   if (IsCertificateSelfSigned(pCertContext, pCertContext->dwCertEncodingType, 0) || IsCACert(pCertContext))
-                { continue;
+            {    
+                if (IsCertificateSelfSigned(pCertContext, pCertContext->dwCertEncodingType, 0))
+                {
+                    if (bAllowUnknownSignature)
+                        return pCertContext;
+                    continue;
+                }
+                if (IsCACert(pCertContext))
+                {
+                    continue;
                 }
                 else
                 {   //end entity cert
-                    break;
+                    return pCertContext;
                 }
             }
+            return NULL;
         }
-        return pCertContext;
     }
     
     static bool DoesSignatureCertContainStoreEKU(_In_ byte* rawSignatureBuffer, _In_ ULONG dataSize)
     {
-        unique_cert_context certificateContext(GetCertContext(rawSignatureBuffer, dataSize));        
+        unique_cert_context certificateContext(GetCertContext(rawSignatureBuffer, dataSize, false));
+        if (certificateContext.get() == NULL)
+            return false; 

```

Comment:
add braces to if statement
```suggestion
        {
            return false; 
        }
```